### ISSUE TITLE
Inconsistency in Package versions when running in Documenter compared to Pluto

### ISSUE THREAD

**Issue by kellertuer** [rikhuijzer/PlutoStaticHTML.jl#147]
First of all, I think this package is quite stable by now, which is nice.

I recently noticed a small difference when rendering the documentation of

https://manoptjl.org/stable/tutorials/GeodesicRegression/

Running the notebook from 

https://github.com/JuliaManifolds/Manopt.jl/blob/master/tutorials/GeodesicRegression.jl

has one major difference:

While running the documentation (with PlutoStaticHTML) used the environment of the docs, most prominently the most recent version of Manopt.jl in my case, running Pluto (with its local environment for the notebook) ran the code with an old version.

Maybe I was not so clever to change the code (in an editor and not in Pluto) to use a new feature. This made PlutoStaticHTML still work fine and render the docs, while when I came back to open the notebook, that one failed.
In reality this would mainly (only) affect breaking changes, of course.

So long story short:
Would it be possible to run notebooks with the pkg environment Pluto has set up for them? That would be neat.

**Comment by rikhuijzer** [comment]
> First of all, I think this package is quite stable by now, which is nice.

That's great to hear! Credits go to the tests. They cover things quite well now.

> Would it be possible to run notebooks with the pkg environment Pluto has set up for them? That would be neat.

Ah yes sure that is possible. If you do that, you'll need different environments and in Julia environments cannot effortlessly be switched without restarting Julia (thanks to the well-known problems with structs and such). Therefore, you need to set `use_distributed` to `true`. This is written in the docs:

> `use_distributed`: Whether to build the notebooks in different processes. By default, this is enabled just like in Pluto and the notebooks are build in parallel. The benefit of different processes is that things are more independent of each other. Unfortunately, the drawback is that compilation has to happen for each process. By setting this option to false, all notebooks are built sequentially in the same process which avoids recompilation. This is likely quicker in situations where there are few threads available such as GitHub Runners depending on the notebook contents. Beware that `use_distributed=false` will not work with Pluto's built-in package manager.

If you do that, then Pluto should automatically pickup built-in pkg environment.

I see that you're currently not running in parallel. Since `EditURL` is now also available via `PlutoStaticHTML`, you could try to pass a list of files to `build_notebooks` to gain a 2x performance improvement (since the GitHub Runners have 2 cores).

**Comment by kellertuer** [comment]
Ah I missed the distributed keyword and will try that.

The Edit URLs are currently broken on my rendering, I am not yet sure why, but that might also be my little-old-code that I hacked a while back.

**Comment by kellertuer** [comment]
Sorry to comment here again but something went very wrong in the newest version, where the only thing I did change was the activation we discussed here

https://manoptjl.org/stable/tutorials/Optimize!/

Do you have an idea why the code end part after the image is not rendered correctly?

**Comment by kellertuer** [comment]
I checked – fixing to 6.0.3 resolves the problem so the page with wrong html was introduced with 6.0.4

**Comment by rikhuijzer** [comment]
The menu is also messed up, right? Or is that something else? This looks bad. I'm sorry for the mess

**Comment by rikhuijzer** [comment]
It's caused by Gumbo adding HTML begin and end stuff because
```
<!DOCTYPE ><HTML><head></head><body>
```
is in the middle of the page

**Comment by kellertuer** [comment]
The menu is messed up because my iterations are in a debug output in code as

```
# 1 F(x) = ...
# 2 F(x) = ...
```

but with code fences not working those are markdown headlines messing up the menu, sure.

**Comment by rikhuijzer** [comment]
I have to work on a paper today and that has priority since it pays the bills. I hope to find time to fix this later this week

**Comment by kellertuer** [comment]
Don't worry – The only thing you changed was the fix in the info box thingy. So I currently just release a patch that pins PlutoStaticHTML to 6.0.3 so the docs are fixed. There is no hurry here :)
